### ISSUE TITLE
[BUG] N4 and occupying the same physical space

### ISSUE THREAD

**Issue by mattcieslak** [PennBBL/qsiprep#57]
```
 caught: 
itk::ExceptionObject (0x7fe35de01fc0)
Location: "unknown" 
File: /Users/mcieslak/projects/ANTs/build/ITKv5-install/include/ITK-5.0/itkImageToImageFilter.hxx
Line: 240
Description: itk::ERROR: N4BiasFieldCorrectionImageFilter(0x7fe35e9c34f0): Inputs do not occupy the same physical space! 
InputImage Direction: 9.9939081e-01 -1.6381450e-05 -3.4900067e-02
2.4508465e-03 9.9756403e-01 6.9713677e-02
3.4813909e-02 -6.9756745e-02 9.9695636e-01
, InputImage_2 Direction: 9.9939081e-01 -1.5328408e-05 -3.4900037e-02
2.4497946e-03 9.9756403e-01 6.9713707e-02
3.4813953e-02 -6.9756738e-02 9.9695636e-01

	Tolerance: 1.0000000e-06
```

To Do:

- [ ] Copy header from b=0 image into the mask image
- [ ] Turn on verbose mode for all N4 nodes

**Comment by cookpa** [comment]
If this happens at the start of execution, then it is likely due to header precision issues. If it happens after a lot of iterations, it's likely due to an ITK bug that was recently fixed

**Comment by cookpa** [comment]
Unfortunately, CopyImageHeaderInformation is not foolproof because it relies on the same ITK code that introduces the deltas.

The only way to fully work around this is to copy the original image into a new image, and use that for all subsequent operations.

eg,

originalInput=original.nii.gz 

workingInput=t1Head.nii.gz 

  CopyImageHeaderInformation $originalInput $workingInput $workingInput 1 1 1

Then run the processing and periodically do

  CopyImageHeaderInformation $originalInput $someImage $someImage 1 1 1

**Comment by mattcieslak** [comment]
I haven't updated the ANTs version in the image since last year. In the case above the error happens right at the beginning, but there have been some cases in antsBrainExtraction where it happens after many steps.

It's probably time for an upgrade. Did the tolerance get changed in the update? Or something about sform to qform? Is the fix in the current ANTs release?

**Comment by cookpa** [comment]
The ITK update fixed an internal error where images would drift out of alignment during filter operations. It doesn't have anything to do with headers.

The header issue has not been fixed, I don't think it can be. Nor have the default tolerances been changed. I did add calls to `CopyImageHeaderInformation` in `antsBrainExtraction.sh`, to try to ensure that the BrainExtractionBrain and BrainExtractionMask had identical headers.

**Comment by cookpa** [comment]
The current release of ANTs (2.3.2) has a bug and you should not use it. The ANTs release system is complicated and has various dependencies on ANTsR etc. We have a collaborator working on improving the release system and providing more regular binaries.

**Comment by mattcieslak** [comment]
How about the current master?

**Comment by cookpa** [comment]
Current master looks good and has the latest ITK 

https://travis-ci.org/ANTsX/ANTs/builds/607441141

**Comment by mattcieslak** [comment]
This got fixed in 0.7.1
